modules/Blockchain.js

```diff
@@ -261,10 +338,62 @@ class Blockchain {
      * Calling this method will subscribe to Blockchain's event which will be
      * emitted globally using globalEmitter.
      * @param event Event to listen to
+     * @param blockchain_id - Blockchain to listen to
      * @returns {number | Object} Event handle
      */
-    async subscribeToEventPermanent(event) {
-        return this.blockchain.subscribeToEventPermanent(event);
+    async subscribeToEventPermanent(event, blockchain_id) {
+        const blockStartConditions = [];
+
+        if (blockchain_id) {
+            const implementation = this.blockchain.find(e => e.config.network_id === blockchain_id);
+
+            const startBlockNumber = await implementation.getCurrentBlock();
+            blockStartConditions.push({
+                blockchain_id,
+                block: { [Op.gte]: startBlockNumber },
+            });
+        } else {
+            const promises = [];
+
+            for (let i = 0; i < this.blockchain.length; i += 1) {
+                const implementation = this.blockchain[i];
+                promises.push(implementation.getCurrentBlock());
+            }
+            const currentBlocks = await Promise.all(promises);
+
+            for (let i = 0; i < currentBlocks.length; i += 1) {
+                const implementation = this.blockchain[i];
+
+                blockStartConditions.push({
+                    blockchain_id: implementation.config.network_id,
+                    block: { [Op.gte]: currentBlocks[i] },
+                });
+            }
+        }
+
+        const that = this;
+        const handle = setInterval(async () => {
+            if (!that.appState.started) {
+                return;
+            }
+
+            const where = {
+                event,
+                finished: 0,
+                [Op.or]: blockStartConditions,
+            };
+
+            const eventData = await Models.events.findAll({ where });
+            if (eventData) {
+                eventData.forEach(async (data) => {
+                    this.emitter.emit(`eth-${data.event}`, JSON.parse(data.dataValues.data));
```

Comment:
This is blockchain specific code, is it possible to move it into specific implementation?